compiler/plugins/target/ROCM/builtins/ukernel/argmax_ukernel.c

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <float.h>
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+
+extern "C" __device__ __attribute__((const)) half __ockl_wfred_max_f16(half);
+extern "C" __device__ __attribute__((const)) float __ockl_wfred_max_f32(float);
+extern "C" __device__ __attribute__((const))
+int64_t __ockl_wfred_min_i64(int64_t);
+extern "C" __device__ __attribute__((const))
+int32_t __ockl_wfred_min_i32(int32_t);
+
+extern "C" __device__ void
+__iree_uk_rocm_argmax_F32I32(float *inputBuffer, size_t input_offset,
+                             int32_t *outputBuffer, size_t output_offset,
+                             size_t reductionSize, uint32_t flag) {
+  uint laneID = __builtin_amdgcn_workitem_id_x();
+  // Set identity value to handle problem non divisible by subgroupSize.
+  float laneMax =
+      laneID >= reductionSize ? -FLT_MAX : inputBuffer[input_offset + laneID];
+  int32_t laneResult = laneID;
+
+  // NOTE: On F32 kernels with clang, reductionSize/blockDim.x has numerical
+  // inaccuracy.
+  uint numBatches = reductionSize / warpSize + 1;
+  for (int i = 1; i < numBatches; ++i) {
+    uint idx = warpSize * i + laneID;
+    float new_in =
+        idx >= reductionSize ? -FLT_MAX : inputBuffer[input_offset + idx];
+    laneResult = new_in > laneMax ? idx : laneResult;
```

Comment:
Should we compare the new `laneMax` with the old value to decide `laneResult` to be consistent? Is `operator>` and `__ocml_fmax_f32` consistent with each other?

Reply:
It should be equivalent, but I think it makes sense to also do something like:
```c
    laneMax = __ocml_fmax_f32(newIn, laneMax);
    laneResult = newIn == laneMax ? idx : laneResult;
```
For better consistency

Reply:
Done! refactored the ukernel to use result of `ocml_fmax` for predicate of `laneResult` for better consistency. 

btw IIRC, the reason I started using _ocml_fmax_f32 instead of `>` is because `>` lowers to a function with `_ocml_fmax_f32` + other unnecessary instructions.